app/src/frontend/building/building-view.tsx

```diff
@@ -14,26 +14,35 @@ import StreetscapeContainer from './data-containers/streetscape';
 import CommunityContainer from './data-containers/community';
 import PlanningContainer from './data-containers/planning';
 import LikeContainer from './data-containers/like';
+import { Building } from '../models/building';
+
+
+interface BuildingViewProps {
+    cat: string;
+    mode: 'view' | 'edit' | 'multi-edit';
+    building: Building;
+    building_like: boolean;
+    user: any;
+    selectBuilding: (building: Building) => void
+}
 
 /**
  * Top-level container for building view/edit form
  *
  * @param props
  */
-const BuildingView = (props) => {
+const BuildingView: React.FunctionComponent<BuildingViewProps> = (props) => {
     switch (props.cat) {
         case 'location':
             return <LocationContainer
                 {...props}
-                key={props.building && props.building.building_id}
```

Comment:
this and other similar lines in this file: the whole component subtree is no longer recreated on building change. This prevent UI-only-related state (like field groups being collapsed or not) from being reset.